src/Compilers/CSharp/Portable/Syntax/SyntaxReplacer.cs

```diff
@@ -179,16 +191,28 @@ private bool ShouldVisit(TextSpan span)
             [return: NotNullIfNotNull(nameof(node))]
             public override SyntaxNode? Visit(SyntaxNode? node)
             {
-                SyntaxNode? rewritten = node;
+                var rewritten = node;
 
                 if (node != null)
                 {
+                    var isReplacedNode = _nodeSet.Remove(node);
```

Comment:
>var isReplacedNode [](http://example.com/codeflow?start=20&length=18)

Consider using `bool` rather for the added `isReplacedNode` declarations. `bool` is almost as short as `var` and provides more value.